stockfish/models.py

```diff
@@ -706,6 +723,33 @@ def get_evaluation(self) -> dict:
             elif splitted_text[0] == "bestmove":
                 return evaluation
 
+    def get_static_eval(self) -> Optional[float]:
+        """Sends the 'eval' command to stockfish to get the static evaluation. The current position is
+           'directly' evaluated -- i.e., no search is involved.
+
+        Returns:
+            A float representing the static eval, unless one side is in check or checkmated,
+            in which case None is returned.
+        """
+
+        fen_position = self.get_fen_position()
+        # Stockfish gives the static eval from white's perspective:
+        compare = 1 if not self.get_turn_perspective() or ("w" in fen_position) else -1
+        self._put(f"position {fen_position}")
```

Comment:
Just for my understanding, why do we need this line?

Reply:
It's a little similar to turn perspective stuff in get_evaluation(), although in this case the 'eval' command returns values from white's perspective (rather than how stockfish normally does it from the side to move's perspective). Basically there's four cases to account for:

- The position is white to move, and the user wants eval from white's side.
- Black to move, user wants eval from white's side.
- White to move, user wants eval from the side who's turn it is.
- Black to move, user wants eval from the side who's turn it is.

For the first two cases, the `not self.get_turn_perspective()` will be true, so compare will = 1 and stockfish's outputted value for the `eval` command remains unchanged. This is good since it's from White's perspective already and the user wants that.

For the last two cases, the value of `compare` is determined by `("w" in fen_position)`. If it's white to move, `compare` will be 1, and Stockfish's `eval` value remains unchanged. It gives from White's perspective, which matches the user's turn perspective requirement since it's white to move. Otherwise if it's black to move, in this case the eval from stockfish needs to be multiplied by -1, to get from black's perspective.

Reply:
Thank you for the explanation. I'm sorry that you had to write so much now, because I was really only concerned with the last line 😅. I don't know if I clicked wrong or if Github always shows so much even if you add the comment for only one line. But my question was why do we have to set the current position again with `self._put(f"position {fen_position}")`?

Reply:
@kieferro Ahh lol, no worries. Yeah I think it's just what Github does, since I've seen it highlight multiple lines before when just one line was being referred to.
Good point about setting the position, it shouldn't be needed. I used get_evaluation as a guide when coding this function, but actually I think the line's unnecessary there as well. I'll update both functions in a new commit.

Reply:
Ah, I see. Sounds good